jerry-core/ecma/operations/ecma-arraybuffer-object.c

```diff
@@ -68,6 +69,11 @@ ecma_op_create_arraybuffer_object (const ecma_value_t *arguments_list_p, /**< li
     }
   }
 
+  if (length + sizeof (ecma_extended_object_t) + JMEM_ALIGNMENT - 1 <= length)
```

Comment:
These expressions might be optimized out by the compiler. I saw such things before. Why not simply check the return value of alloc? The ecma_arraybuffer_new_object could return with NULL if the allocation is not successful.